Reproduccion de canciones

VctPerez opened the pull request.

VctPerez committed:
reproduccion de cancion en boton y de lista

VctPerez committed:
refactorizacion pasar canciones en la  cola

VctPerez committed:
Mantener orden de cola + animaciones de reproductor

VctPerez committed:
Evento de finalizacion de lista

VctPerez committed:
Reinicio de la cola de una misma playlist

VctPerez committed:
Merge branch 'main' into playback-songs

VctPerez committed:
cortar cola al empezazr cancion random

VctPerez committed:
Audio entre pantallas , pero se corta entre estas

VctPerez committed:
No reproducir en otra vista si esta pausado

VctPerez committed:
Refactor song controller con objeto track

VctPerez committed:
Info de cancion + progresion en musicplayer

VctPerez committed:
Botones musicplayer funcionales

Falta corregir que cuando se pasa de cancion se deja de hacer la animacion de la playlist

VctPerez committed:
Fix control de animaciones de reproductores

VctPerez committed:
volumen funcional

VctPerez committed:
modo aleatorio funcional

VctPerez committed:
fix random en otra lista + boton mute

VctPerez committed:
cancion en bucle + barra de progresion interactuable

VctPerez committed:
Merge branch 'main' into playback-songs

VctPerez committed:
Correccion de errores al mergear

JoseBueno30 reviewed (commented):
Si reproduces una canción desde la barra de búsqueda, después no te deja reproducir canciones de la playlist.

También he visto que si una canción no tiene preview se reproduce la siguiente en su lugar, entiendo que es feature pero he encontrado una cosa un poco rara, tenia una playlist cuya ultima canción si tenia previw y la penúltima no, pero al borrar la ultima canción la penúltima seguía reproduciéndola incluso después de recargar la pagina.

Esto no se si es cosa del componente de MusicPlayer pero a veces pasa esto 
![image](https://github.com/JoseBueno30/Dutify/assets/124668506/49e45917-2dbb-464b-9557-9e3dc5aa5a27)

JoseBueno30 committed:
fix feedback listas seguidas

JoseBueno30 commented:
Había un pequeño fallo con el feedback de añadir una playlist a favoritas porque se había cambiado el nombre de playlist a playList en algunos sitios y no en otros, he arreglado  solo eso y lo pusheo

adriantd3 reviewed (changes requested):
Esta increible victor.

Solo digo lo que ha puesto jose, lo de que hay canciones que se raya la imagen porque no es cuadrada (entiendo que esto es tema del music player). 

Lo otro que menciono es lo que ya te dije sobre las canciones que no tienen preview. No tiene sentido que aparezcan canciones con las que no puedes interaccionar. Yo haría que no saliesen directamente aunque entiendo que eso daría un monton de problemas.
Por lo menos pondría un popup indicando que la canción no tiene preview y que por tanto, se reproducira la siguiente que sí tiene.

JoseBueno30 committed:
arreglo carga playlist

JoseBueno30 committed:
fix

JoseBueno30 committed:
try catch loading

JoseBueno30 committed:
fix

JoseBueno30 committed:
botones loop y random responsive

JoseBueno30 committed:
beta boton play de canciones feedback

JoseBueno30 committed:
cancion que se reproduce se muestra en el botón

hay un pequeño bug, si una canción se reproduce desde la búsqueda y también está en la lista esta se ve como reproducida también.

JoseBueno30 committed:
fix

JoseBueno30 committed:
undefined trackObject fix

JoseBueno30 committed:
fix busqueda

JoseBueno30 committed:
accesibilidad

JoseBueno30 committed:
popup feedback

VctPerez committed:
Merge branch 'main' into playback-songs

VctPerez committed:
pausar desde songbutton + bug random songbutton

Bug random songbutton: el boton de la cancion no reproducia su cancion cuando el modo aleatorio estaba activado.

VctPerez committed:
fix reproducir lista post cancion busqueda

No dejaba reproducir la lista despues de haber reproducido una cancion desde el desplegable de la busqueda en la misma pantalla de la lista

VctPerez committed:
Fix reproduccion de canciones eliminadas de lista

VctPerez committed:
fix bug al eliminar en random + random rediseñado

VctPerez committed:
Remake inLoop

JoseBueno30 committed:
songButton funciona correctamente

-Ahora pausa la canción y la continua reproduciendo desde el mismo punto
-Habia que hacer click dos veces para poder reproducir, era por un fallo en las condiciones del clickHandler porque se comprobaba el estado del ocmponente que siempre se iniciaba a false.

JoseBueno30 committed:
recarga al añadir cancion a playlist

JoseBueno30 committed:
elimninar playlist arreglado

JoseBueno30 commented:
- [x] Al poner el modo en bucle, si reproduces otra canción de la playlist (he probado con la última) parece que se desactiva el bucle
- [x] Hay que actualizar la playlist cuando ser recarge la lista, he implementado que al ñadir una canción desde el despegable la pagina se recarge pero esto afecta a la queue y no deja reproducir esa canción. 
- [x] La playlist con el modo aleatorio y en bucle a la vez creo que no termina de funcionar (al menos parece que se para despues de reproducir algunas canciones).
- [x] Si activas el modo random y despues lo desactivas sigue reproduciendo de forma aleatoria.
- [x] Si activas el modo random al clickar sobre una canción se reproduce una de forma aleatoria de la playlist.
- [x] Al activar el modo random, ir a cualquier otra pagina y después volver a una playlist el botón aparece como activo pero en realidad no lo está, con el loop no ocurre, en ese casi si que se mantiene activo.

JoseBueno30 committed:
fix añadir a playlist

VctPerez committed:
Fix no random despues de random

VctPerez committed:
fix bug botones random y shuffle + random en otra lista

JoseBueno30 committed:
arreglo logitud boton e info de canciones

JoseBueno30 committed:
fix

JoseBueno30 committed:
añadir y eliminar modifica la queue

VctPerez committed:
Fix reproduccion de cancion añadida

VctPerez committed:
Merge branch 'main' into playback-songs

JoseBueno30 committed:
fix tamaño imagen lista

JoseBueno30 committed:
navegacion playlist mejorada

JoseBueno30 committed:
estilos adecuados cuando el boton se selecciona con flechas

adriantd3 reviewed (changes requested):
- [x] He probado a hacer lo siguiente y falla: Entrar una playlist, reproducir una canción, activar modo random, dejar que pase a la siguiente y clickar varias veces el boton de cancion anterior. Esto hace que no retroceda a mas canciones anteriores y que ademas no puedas saltar a la siguiente. Además, cuando se acaba la preview de la canción no pasa a la siguiente.
- [x] Cuando una canción no tiene preview, salta a la siguiente pero no te deja volver a la anterior, se resetea.
- [x] Con una playlist de 3 canciones y en modo random, he contado que puedes reproducir 4 antes de que te corte. Esto lo he probado pulsando el boton de saltar cancion y dejandolo normal.
- [x] Cuando se acaba la playlist, la ultima canción se queda rayada. Si quieres reproducirla de nuevo tienes que darle al boton de reproducir de songButton 2 veces o al boton de reproducir de abajo. En general en este estado esta bugueado, no te deja volver a la cancion anterior o pasar a una siguiente con el modo random. La unica solucion es darle al boton de reproducir lista.
- [x] Si añades una nueva canción DESDE EL BOTON AÑADIR CANCNIÓN mientras se esta reproduciendo la lista, nunca llegará a esa nueva canción, ni con modo random o bucle. Además si le das a reproducir, se reproduce, pero no sigue a otras canciones de la playlist.

VctPerez committed:
Fix bug cancion anterior

VctPerez committed:
fix anterior sin preview

VctPerez committed:
Quitar cancion actual de la cola random

VctPerez committed:
rework aleatorio + fix fin de cola/cancion

VctPerez committed:
Fix reproducir cancion añadida desde boton añadir

adriantd3 reviewed (approved):
Todo finisimo.👍

JoseBueno30 committed:
Merge branch 'main' into playback-songs

JoseBueno30 committed:
musicPlayer toString bug fix

JoseBueno30 committed:
fix accesibilidad boton añadir
